sdk/lib/atl/atlbase.h

```diff
@@ -662,6 +665,127 @@ class CAtlDllModuleT : public CAtlModuleT<T>
     }
 };
 
+
+template <class T>
+class CAtlExeModuleT : public CAtlModuleT<T>
+{
+public:
+    DWORD m_dwMainThreadID;
+    //DWORD m_dwTimeOut;
+    //DWORD m_dwPause;
+    //bool m_bDelayShutdown;
+
+    CAtlExeModuleT()
+    {
+        m_dwMainThreadID = ::GetCurrentThreadId();
+    }
+
+    ~CAtlExeModuleT()
+    {
+    }
+
+    int WinMain(int nShowCmd)
+    {
+        HRESULT hr = T::InitializeCom();
+        if (FAILED(hr))
+            return hr;
```

Comment:
Shouldn't this function return 0/1 or something?

Reply:
Usually 0 means success, so S_OK maps perfectly here.

Reply:
But whatever the value of 'hr' is in this failure case wouldn't :p